subprojects/core/src/main/java/org/gradle/internal/progress/BuildProgressFilter.java

```diff
@@ -109,8 +113,12 @@ public void beforeComplete() {
 
     @Override
     public void buildFinished(BuildResult result) {
-        if (result.getGradle() == gradle) {
-            gradle = null;
+        if (result.getGradle() == rootGradle) {
+            rootGradle = null;
         }
     }
+
+    private boolean isRootTaskGraph(TaskExecutionGraph graph) {
```

Comment:
:o: I'd probably add a Javadoc comment here to explain what "root" means specifically.